utils/test_results.py

```diff
@@ -64,38 +72,42 @@ def aggregate_test_results(
             branch=branch
         )
 
-    failure_rates_queryset = (
-        pass_failure_error_test_instances.values("test")
-        .annotate(
-            failure_rate=Avg(
-                Case(
-                    When(outcome="pass", then=Value(0.0)),
-                    When(outcome__in=["failure", "error"], then=Value(1.0)),
+    totals = DailyTestRollup.objects.filter(repoid=repoid, date__gt=time_ago)
+
+    print([(t.pass_count, t.fail_count) for t in totals], branch)
+
+    if branch is not None:
+        totals = totals.filter(branch=branch)
+
+    commits_where_fail_sq = (
+        totals.filter(test_id=OuterRef("test_id"))
+        .annotate(v=Distinct(Unnest(F("commits_where_fail"))))
+        .values("v")
+    )
+    latest_duration_sq = (
+        totals.filter(test_id=OuterRef("test_id"))
+        .values("last_duration_seconds")
+        .order_by("-latest_run")[:1]
+    )
+
+    aggregation_of_test_results = totals.values("test").annotate(
+        failure_rate=(
+            Cast(Sum(F("fail_count")), output_field=FloatField())
```

Comment:
Does the Cast(Sum(F("fail_count"))) get cached when we do this calc twice? I don't have an idea of how we'd simplify, mostly curious if you knew

Reply:
i would expect postgres to take care of this in a performant way

Reply:
Fair enough